### ISSUE TITLE
ABCL-specific code is incompatible with JEP 396 & 403

### ISSUE THREAD

**Issue by se-mz** [cl-plus-ssl/cl-plus-ssl#182]
Simple example:
```
> (lisp-implementation-type)
"Armed Bear Common Lisp"
> (lisp-implementation-version)
"1.9.2"
"OpenJDK_64-Bit_Server_VM-Gentoo-17.0.6+10"
"amd64-Linux-4.19.289-gentoo"
> (drakma:http-request "https://example.com")
WARNING: cl+ssl:stream-fd: all approaches failed. stream: #<TWO-WAY-STREAM {7FFB692D}>, jvm-version: 17, internal input-stream: #<java.io.BufferedInputStream java.io.BufferedInputStream@3555.... {11434B66}>, errors:
Java exception 'java.lang.reflect.InaccessibleObjectException: Unable to make field protected volatile java.io.InputStream java.io.FilterInputStream.in accessible: module java.base does not "opens java.io" to unnamed module @3235900c'.
Java exception 'java.lang.reflect.InaccessibleObjectException: Unable to make field protected volatile java.io.InputStream java.io.FilterInputStream.in accessible: module java.base does not "opens java.io" to unnamed module @3235900c'.
Java exception 'java.lang.reflect.InaccessibleObjectException: Unable to make field protected volatile java.io.InputStream java.io.FilterInputStream.in accessible: module java.base does not "opens java.io" to unnamed module @3235900c'.
; Evaluation aborted on #<TYPE-ERROR {5D390720}>.
```
Works fine under Java 11. The entire purpose of JEP 396 & 403 is basically to prevent what cl+ssl is doing here, so I doubt that this is resolvable on the library end. It can however be worked around by the user by adding `--add-opens java.base/java.io=ALL-UNNAMED --add-opens java.base/sun.nio.ch=ALL-UNNAMED` to the JVM command-line; it might be useful to document this. Maybe there's also something that can be done on the ABCL side, not sure.

**Comment by avodonosov** [comment]
Thank you for the report.

Does switching Lisp BIO, like below, work?

```common-lisp
(let ((cl+ssl:*default-unwrap-stream-p* nil))
   (drakma:http-request "https://example.com"))

```

**Comment by se-mz** [comment]
It doesn't, same issue.

**Comment by avodonosov** [comment]
Same issue? Strange, the binding tells cl+ssl to not try retrieving file descriptor from the socket. 

Are you sure?

**Comment by avodonosov** [comment]
Ah, I see what's going on.

This pull request to drakma sits unmerged: https://github.com/edicl/drakma/pull/120

Without it drakma explicitly calls `(stream-fd )`, so `cl+ssl:*default-unwrap-stream-p*` has no effect.

**Comment by se-mz** [comment]
Ran another test after applying the linked Drakma patch; both BIO types work now, although the unwrapped one outputs warnings.

**Comment by avodonosov** [comment]
Yes, with default value of `cl+ssl:*default-unwrap-stream-p* = t` cl+ssl tries to call `(stream-fd socket)` which on ABCL with new Java prints the warn and returns `nil`, so cl+ssl fallbacks to use the `socket` parameter as is, which is a lisp socket stream, so it result Lisp BIO is used.

https://github.com/cl-plus-ssl/cl-plus-ssl/blob/17d5cdd65405f1d26e26f3e875e70027d0c8eedb/src/streams.lisp#L231

Note, the drakma patch is merged now.

**Comment by avodonosov** [comment]
What remains: documenting.

The warning printed is a kind of documentation already. But it may be good to explain in the "Lisp BIO or Socket BIO" section of the main README that extracting the file descriptor may not work on some platforms.

Also, the docstring of stream-fd says "The STREAM's file descriptor as an integer, if known / implemented for the current lisp. Otherwise the STREAM itself.". But in reality the ABCL version returns `nil` on failure.

Either the ABCL version of the function or the docstring (but that would be an API change). BTW, of ABCL version returned the stream itself, drakma would work without the patch.
